### DISCUSSION TITLE
Android APP error when conecting to vaultwarden : "Exception message: Connection failure"

### DISCUSSION THREAD

**Discussion by firesixtieone** [dani-garcia/vaultwarden#4843] [Q&A] [answered]
### Subject of the issue
Android APP not connection to vaultwarden : "Exception message: Connection failure"

I've just update from [1.30.5](https://github.com/dani-garcia/vaultwarden/releases/tag/1.30.5) 
I've made a snapshot of my VM before update, and I tries to reconnect but the same thing appen... I then re-update my vaultwarden to the latest version.

Web and Windows APP working perfectly.

Thanks in advance 

My environment (Generated via diagnostics page)
* Vaultwarden version: v1.32.0
* Web-vault version: v2024.6.2b
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: n/a
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.46.0
* Clients used: 
* Reverse proxy and version:  Apache 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_max_note_size": 10000,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*******************",
  "domain_origin": "*****://*******************",
  "domain_path": "",
  "domain_set": true,
  "duo_context_purge_schedule": "30 * * * * *",
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "duo_use_iframe": false,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "http_request_block_non_global_ips": true,
  "http_request_block_regex": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "increase_note_size_limit": false,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "*************",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "*************************",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "************",
  "smtp_password": null,
  "smtp_port": 25,
  "smtp_security": "off",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

    ========================================================================================
-->

* vaultwarden version: 

* Install method: docker (with podman)

* Clients used: web vault, desktop, Android, iOS

* Reverse proxy and version: apache

* Other relevant details:

### Steps to reproduce
Open the native android bitwarden app, type on "bitwarden.com" then select "self-hosted" type in my URL of vaultwarden instance in the first field "Server's URL" then save.
Type my email in the corresponding field, then press next.
The error is here

### Expected behaviour
Do not have the error and type my password to login

### Actual behaviour
Error "Exception message: Connection failure" before typing password

**Comment by BlackDex** [comment]
If you can access your web-vault, then you should also be able to acces it with any client.

Check the logs of Vaultwarden, and if there are no logs of the attempt there, check your apache logs, if there are also no logs there, then something is wrong with the client.

Double-check the FQDN you provided like http or https etc..

**Reply by firesixtieone** [reply]
The FQDN is wright (Triple check...)

I already tried to delete the cache, delete the data of the app, then delete the app, reboot the phone and install it again and same result

I tested with my wife phone and it's the same result. (Bitwarden has never been installed on her phone) 

As for the log : 

I have this in /var/log/apache2/bitwarden-error.log  (Wich is not good I presume, but I can't see anything relevant in diagnostic...)

![bitwarden-error](https://github.com/user-attachments/assets/9b7235c7-74d3-45d0-929e-cbc62fb47f76)

I have this in /var/log/apache2/bitwarden-access.log (I do not see any "android" request if i'm not wrong...)

![bitwarden-acces](https://github.com/user-attachments/assets/765d7fc4-4771-4bfb-8ebc-3846aaefca88)

Maybe there is an another log file inside the container ? I do not know how to access it thow

**Reply by BlackDex** [reply]
You should at least see something like this.
```
12/Aug/2024:18:19:39 +0000] "POST /identity/connect/token HTTP/1.1" 200 2433 "-" "Bitwarden_Mobile/2024.7.1 (Android 14; SDK 34; Model Pixel 7)"
12/Aug/2024:18:19:40 +0000] "GET /api/accounts/revision-date HTTP/1.1" 200 13 "-" "Bitwarden_Mobile/2024.7.1 (Android 14; SDK 34; Model Pixel 7)"
12/Aug/2024:18:19:40 +0000] "GET /api/sync HTTP/1.1" 200 417479 "-" "Bitwarden_Mobile/2024.7.1 (Android 14; SDK 34; Model Pixel 7)"
12/Aug/2024:18:19:41 +0000] "GET /api/config/ HTTP/1.1" 200 294 "-" "Bitwarden_Mobile/2024.7.1 (Android 14; SDK 34; Model Pixel 7)"
12/Aug/2024:18:19:43 +0000] "GET /api/auth-requests/ HTTP/1.1" 200 83 "-" "Bitwarden_Mobile/2024.7.1 (Android 14; SDK 34; Model Pixel 7)"
12/Aug/2024:18:19:43 +0000] "GET /api/accounts/revision-date HTTP/1.1" 200 13 "-" "Bitwarden_Mobile/2024.7.1 (Android 14; SDK 34; Model Pixel 7)"
12/Aug/2024:18:19:44 +0000] "GET /api/config/ HTTP/1.1" 200 294 "-" "Bitwarden_Mobile/2024.7.1 (Android 14; SDK 34; Model Pixel 7)"
```

**Reply by BlackDex** [reply]
Also the error logs shows it sometimes has issues connecting to the container for some reason.

**Reply by firesixtieone** [reply]
I've tried several time, there is no way I can have some log like you ^^ 
When I try on firefox on my smartphone I can access the login page and log in normally.

I notice that the http were not redirect to https it goes to the default apache page (But i shoudl't mater because I put the adress with https ...)

There is my apache config if it can help : 
![vaultwarden-apache-conf](https://github.com/user-attachments/assets/416ba5f8-e3c5-4010-b8f4-63c59f3367a4)

I launch the container with this command : 
podman run -d --name vaultwarden -v /var/vaultwarden/:/data/:Z -p 8080:80 docker.io/vaultwarden/server:latest

Maybe I can mp you my FQDN so you can try on your android device to see if it make any difference ?

I am loosing my mind to this error ... 

Thanks for helping me :)

**Reply by BlackDex** [reply]
I would try to fully stop and clear cache and data of the Bitwarden app and start again.

Also, only fill in the first input box, not the rest.

**Reply by firesixtieone** [reply]
Yes I fill in the first input box only with https://vault.mydomain.fr

Like I said, I already forced stop, empty cache and data.
Uninstalled the app, reboot and re-install the app

Same error on a completely different phone ...

**Reply by BlackDex** [reply]
Are you on wifi or mobile data? Does that make a difference?
Do you use an add blocker? Or maybe DoH/Private DNS which might cause issues.

**Reply by firesixtieone** [reply]
I've tried my Home network (No specific DNS or other) My work network same not particularity
And Mobile data
Same result everytime

**Comment by firesixtieone** [comment, marked as answer]
Well foudn it now !

It's about the certificate and the Full chain certificate.

Turn out that the intermediate certificate (The certificate of Gandi, the one who provide the certificate) was expired or something

I re-download it and change it in my apache config and now it works ! 

Test SSL with Qualys SSL Lab :

Before : 

![before](https://github.com/user-attachments/assets/3b3df0a0-8ce7-4f37-b2c7-deb105b10bbd)

And after : 

![after](https://github.com/user-attachments/assets/435cdb19-a4c7-4c48-9dd6-8e99ffa63511)
